src/Lesson4_2.java

```diff
@@ -0,0 +1,65 @@
+public class Lesson4_2 {
+    public static void main(String[] args) {
+        task1();
+        task2();
+        task3();
+        task4();
+    }
+
+    public static int[] generateRandomArray() {
+        java.util.Random random = new java.util.Random();
+        int[] arr = new int[30];
+        for (int i = 0; i < arr.length; i++) {
+            arr[i] = random.nextInt(100_000) + 100_000;
+        }
+        return arr;
+    }
+
+    public static void task1() {
+        System.out.println("Task 1");
+        int expenseDuringMonth = 0;
+        int[] expense = generateRandomArray();
+        for (int i = 0; i < 30; i++){
+            expenseDuringMonth += expense[i];
+        }
+        System.out.println("Сумма трат за месяц составила " + expenseDuringMonth + " рублей");
+        System.out.print("\n");
+    }
+
+    public static void task2() {
+        System.out.println("Task 2");
+        int maxExpense = 0, minExpense = 10000000;
+        int[] expense = generateRandomArray();
+        for (int i = 0; i < 30; i++){
```

Comment:
откуда 30? надо длину массива использовать, а не числа "с потолка":
```suggestion
        for (int i = 0; i < expense.length; i++){
```